src/c++/perf_analyzer/command_line_parser.cc

```diff
@@ -1482,6 +1511,61 @@ CLParser::ParseCommandLine(int argc, char** argv)
           params_->profile_export_file = profile_export_file;
           break;
         }
+        case 59: {
+          params_->using_periodic_concurrency_range = true;
+          std::string arg = optarg;
+          size_t pos = 0;
+          int index = 0;
+          while (pos != std::string::npos) {
+            size_t colon_pos = arg.find(":", pos);
+            if (index > 2) {
+              Usage(
+                  "Failed to parse --periodic-concurrency-range. The value "
+                  "does not match <start:end:step>.");
+            }
+            int64_t val;
+            if (colon_pos == std::string::npos) {
```

Comment:
This code snippet exists in a few places. Is it time to refactor this into a separate method?